llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -1224,6 +1224,42 @@ class VPPhiAccessors {
     return getAsRecipe()->getNumOperands();
   }
 
+  /// Returns an interator range over the incoming values.
+  VPUser::const_operand_range incoming_values() const {
+    return make_range(getAsRecipe()->op_begin(),
+                      getAsRecipe()->op_begin() + getNumIncoming());
+  }
+
+  using const_incoming_block_iterator =
+      mapped_iterator<detail::index_iterator,
+                      std::function<const VPBasicBlock *(size_t)>>;
```

Comment:
llvm::functtion_ref?

Reply:
I don't think llvm::function_ref will work, as we need to capture the enclosing object